src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -16,6 +16,25 @@ public static bool IsScript(this SyntaxTree syntaxTree)
             return syntaxTree.Options.Kind != SourceCodeKind.Regular;
         }
 
+        /// <summary>
+        /// Get the effective severity of a diagnostic based on the editor config
+        /// or ruleset configuration of the diagnostic id.  If none configured, return default.
+        /// </summary>
+        public static ReportDiagnostic GetEffectiveSeverity(this SyntaxTree syntaxTree, string diagnosticId, Compilation compilation)
+        {
+            if (syntaxTree.DiagnosticOptions.TryGetValue(diagnosticId, out var syntaxTreeSeverity))
+            {
+                return syntaxTreeSeverity;
+            }
+
+            if (compilation.Options.SpecificDiagnosticOptions.TryGetValue(diagnosticId, out var compilationSeverity))
+            {
+                return compilationSeverity;
+            }
+
+            return ReportDiagnostic.Default;
```

Comment:
@sharwell do we have code that already does this somewhere?

Reply:
I don't know of one but @jasonmalinowski may have added one when moving to the new features

Reply:
We have [DiagnosticDescriptor.GetEffectiveSeverity](http://source.roslyn.io/#Microsoft.CodeAnalysis/Diagnostic/DiagnosticDescriptor.cs,205), but that API only with on compilation options (pre-editorconfig world).